src/index.ts

```diff
@@ -31,17 +21,19 @@ const waitForExpect = function waitForExpect(
   timeout = defaults.timeout,
   interval = defaults.interval
 ) {
-  const startTime = now();
+  const maxTries = Math.ceil(timeout / interval);
```

Comment:
I would probably throw an error here if `interval === 0`. Otherwise it's going to cause an infinite loop.

```suggestion
  if (interval === 0) {
    throw Error("Interval can't be 0");
  }

  const maxTries = Math.ceil(timeout / interval);
```

Reply:
Good point. The cases are
```js
Math.ceil(1 / 0) // => Infinity --- will retry forever
Math.ceil(0 / 0) // => NaN --- will also retry forever, because tries > NaN is always false
```
That is probably not what the caller intended.

An alternative to raising an error would be to clamp `interval` at 1:
```
if (interval < 1) interval = 1;
```
but that might also not be what the caller intended, because if the check is very fast, `waitForExpect` might not wait as long as it did before this change.

I am leaning toward making it an error. I've pushed up a change along the lines of what you suggest, but I figure we can also exclude negative intervals while we are at it.

This could arguably be considered a breaking change (would suggest a major version bump), because anyone who had set their interval to zero would get an error.

Reply:
I say default it to 1. Nobody expects it to be so precise anyway.

Reply:
> I say default it to 1

That's also fine by me, for the record 👍. I'm happy to make that change in this PR (or the maintainers can), but since we've now heard back from @lgandecki I'll wait for input.

Reply:
I agree with the default of 1 ms. 

If I understand correctly setTimeout(fn, 0) just means "next tick", while setTimeout(fn, 1) means "next tick, but not sooner than 1 ms". It's hard to imagine that "next tick" happening noticeably sooner than 1ms anyway, for these two calls to produce different results. Making software or tests rely on such unreliable assumptions (especially between dev machines and CI envs) would be very strange. 

People needing such level of precision, as Kent mentions, probably shouldn't use this tiny tool (which is built for situations that are hard to define with precision), or JS for that matter :-)

The rest looks good to me and I'm happy to merge. This is a great change and will prevent many sorts of existing and future problems.

Reply:
Great 👍 . I've pushed 329551e with that change.